### PR TITLE
Prachi branch pin and unpin notes

### PR DESCRIPTION
**Pull request by prachi2023**

### CODE REVIEW

File: src/main/java/seedu/duke/command/PinCommand.java [hunk 1]
```diff
@@ -48,6 +50,30 @@ public PinCommand(String title) {
 
     @Override
     public String execute() {
-        return null;
+        Note note = new Note("", "", false);
```
Comment (Chongjx) [inline src/main/java/seedu/duke/command/PinCommand.java:53]: Is it necessary to create a new Note object here?

File: src/main/java/seedu/duke/command/PinCommand.java [hunk 2]
```diff
@@ -48,6 +50,30 @@ public PinCommand(String title) {
...
+        boolean noteExists = false;
+        if (isPinByIndex) {
+            try {
+                note = notebook.getNotes().get(index);
+            } catch (IndexOutOfBoundsException exception) {
+                return "Note with this index does not exists in the notebook";
```
Comment (Chongjx) [inline src/main/java/seedu/duke/command/PinCommand.java:59]: Maybe can make this a static final string as COMMAND_UNSUCCESSFUL_MESSAGE

File: src/main/java/seedu/duke/util/Parser.java [hunk 3]
```diff
@@ -567,11 +567,37 @@ private Command prepareEditNote(String userMessage) {
...
+        try {
+            ArrayList<String[]> splitInfo = splitInfoDetails(userMessage);
+
+            for (String[] infoDetails : splitInfo) {
+                String prefix = infoDetails[0].toLowerCase();
+                ExceptionType exception;
+                switch (prefix) {
+                case PREFIX_TITLE:
+                    exception = ExceptionType.EXCEPTION_MISSING_TITLE;
+                    title = checkBlank(infoDetails[1],exception);
```
Comment (Chongjx) [inline src/main/java/seedu/duke/util/Parser.java:584]: A space after the comma

**Review by Chongjx — APPROVED** [review]
Looks good! Just some minor changes.

File: src/main/java/seedu/duke/util/Parser.java (hunk 3 above)
Comment (R-Ramana) [inline src/main/java/seedu/duke/util/Parser.java:584]:
```suggestion
                    title = checkBlank(infoDetails[1], exception);
```

**Review by R-Ramana — APPROVED** [review]

**Pull request merged**